packages/backend/src/services/ProjectService/ProjectService.ts

```diff
@@ -4019,4 +4045,167 @@ export class ProjectService extends BaseService {
+                              spaceSummary.uuid,
+                          ),
+                      }),
+                  )
+                : user.ability.cannot(
+                      'promote',
+                      subject('SavedChart', {
+                          organizationUuid,
+                          projectUuid,
+                      }),
+                  );
+            if (userDontHaveAccess) {
+                throw new ForbiddenError(
+                    `You must have the right permission on ${context} to promote this chart`,
+                );
+            }
+        };
+
+        const promotedChart = await this.savedChartModel.get(
+            chartUuid,
+            undefined,
+        );
+        const { organizationUuid, projectUuid } = promotedChart;
+        if (promotedChart.dashboardUuid)
+            throw new Error(`We can't promote charts within dashboards`);
+        const space = await this.spaceModel.getSpaceSummary(
+            promotedChart.spaceUuid,
+        );
+
+        await checkPermissions(
+            organizationUuid,
+            projectUuid,
+            space,
+            'this chart and project',
+        );
+
+        const { upstreamProjectUuid } = await this.projectModel.getSummary(
+            projectUuid,
+        );
+
+        if (!upstreamProjectUuid)
+            throw new Error('This chart does not have an upstream project');
+
+        const { slug } = promotedChart;
+        if (!slug) {
+            // We could create a new slug here on the fly if needed
+            throw new Error('This chart does not have a valid identifier');
+        }
+        const existingUpstreamCharts = await this.savedChartModel.find({
+            projectUuid: upstreamProjectUuid,
+            slug,
+        });
+
+        if (existingUpstreamCharts.length === 0) {
+            // There are no chart with the same slug, we create the chart, and the space if needed
+            // We only check the org/project access, we will create a new space if needed
+            await checkPermissions(
+                organizationUuid,
+                upstreamProjectUuid,
+                undefined,
+                'the upstream project',
+            );
+
+            let newSpaceUuid: string;
+            const existingSpace = await this.spaceModel.find({
+                projectUuid: upstreamProjectUuid,
+                slug: space.slug,
+            });
+            if (existingSpace.length === 1) {
+                // We have an existing space with the same slug
+                await checkPermissions(
+                    organizationUuid,
+                    upstreamProjectUuid,
+                    existingSpace[0],
+                    'the upstream space and project',
+                );
+                newSpaceUuid = existingSpace[0].uuid;
+            } else {
+                // We have 0 or more than 1 space with the same slug
+                await checkPermissions(
```

Comment:
We should check if the user has access to create spaces in the upstream uproject